driver-dse/src/main/java/com/datastax/driver/graph/GraphData.java

```diff
@@ -0,0 +1,215 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.datastax.driver.core.exceptions.DriverException;
+
+/**
+ * A result entity containing a graph query's result, wrapping a Json result.
+ */
+public class GraphData {
+
+    private JsonNode jsonNode;
+
+    private ObjectMapper objectMapper;
+
+    private Object key;
+
+    GraphData(Object key, JsonNode jsonNode, ObjectMapper objectMapper) {
+        this.jsonNode = jsonNode;
+        this.objectMapper = objectMapper;
+        this.key = key;
+    }
+
+    /**
+     * API
+     */
+
+    /**
+     * This method allows digging through the result received from the graph server.
+     *
+     * @param keyOrIndex The key to get the value from, if keyOrIndex is a String. The index of the Array, if it is an Integer.
+     * @return A new GraphData object containing the encapsulated result.
+     */
+    private GraphData get(Object keyOrIndex) {
+        if (keyOrIndex == null) {
+            throw new DriverException("You must provide a valid key or index identifier in a get() call, 'null' is not valid.");
+        }
+        if (this.jsonNode == null) {
+            return new GraphData(keyOrIndex, null, this.objectMapper);
+        }
+        JsonNode jsN;
+        if (keyOrIndex instanceof Integer) {
+            jsN = this.jsonNode.get((Integer)keyOrIndex);
+        } else {
+            jsN = this.jsonNode.get((String)keyOrIndex);
+        }
+        return new GraphData(keyOrIndex, jsN, this.objectMapper);
+    }
+
+    public GraphData get(String key){
```

Comment:
One should clarify that get(String) applies to JSON objects and get(int) to JSON arrays - calling the wrong one returns null (took me a while to figure out).